guides/asynchronous-tasks/readme.md

```diff
@@ -300,57 +300,59 @@ end
 
 ## Timeouts
 
-You can wrap asynchronous operations in a timeout. This ensures that malicious services don't cause your code to block indefinitely.
+You can wrap asynchronous operations in a timeout. This allow you to put an upper bound on how long the enclosed code will run vs. potentially blocking indefinitely. If the enclosed code hasn't completed by the timeout, it will be interrupted with an {ruby Async::TimeoutError} exception.
 
 ~~~ ruby
 require 'async'
 
 Async do |task|
 	task.with_timeout(1) do
-		task.sleep 100
+		sleep 100
 	rescue Async::TimeoutError
-		puts "I timed out!"
+		puts "I timed out 99 seconds early!"
 	end
 end
 ~~~
 
-### Reoccurring Timers
+### Periodic Timers
 
-Sometimes you need to do some periodic work in a loop.
+Sometimes you need to do some recurring work in a loop. Often it's best to measure the periodic delay end-to-start, so that your process always takes a break between iterations and doesn't risk spending 100% of its time on the periodic work. In this case, simply call {ruby sleep} between iterations:
 
 ~~~ ruby
 require 'async'
 
 Async do |task|
-	while true
+	loop do
```

Comment:
Minor: Perhaps we can keep `while true`, since no block will be created by that construction?

Reply:
I am fine with either option. Honestly, Ruby should probably just optimise `loop do...end`.